pandas/core/internals/blocks.py

```diff
@@ -1850,10 +1850,39 @@ def get_values(self, dtype: DtypeObj | None = None) -> np.ndarray:
     def values_for_json(self) -> np.ndarray:
         return self.values
 
-    def delete(self, loc) -> Block:
-        values = np.delete(self.values, loc, 0)
-        mgr_locs = self._mgr_locs.delete(loc)
-        return type(self)(values, placement=mgr_locs, ndim=self.ndim)
+    def delete(self, loc) -> list[Block]:
+        if not is_list_like(loc):
+            loc = [loc]
+
+        if self.ndim == 1:
+            values = np.delete(self.values, loc)
+            mgr_locs = self._mgr_locs.delete(loc)
+            return [type(self)(values, placement=mgr_locs, ndim=self.ndim)]
+
+        if np.max(loc) >= self.values.shape[0]:
+            raise IndexError
+
+        # Add one out-of-bounds indexer as maximum to collect
+        # all columns after our last indexer if any
+        loc = np.concatenate([loc, [self.values.shape[0]]])
+        mgr_locs_arr = self._mgr_locs.as_array
+        new_blocks: list[Block] = []
+
+        previous_loc = -1
+        for idx in loc:
```

Comment:
this reminds me of `BlockManager._slice_take_blocks_ax0` with `only_slice=True` if we passed e.g. `slice_or_indexer = np.delete(np.arange(length), loc))`.  is it worth trying to de-duplicate?

Reply:
Not sure, this seemed to do something different than what we need. Tried to get it to work a while back, but didn't get it to work